docs/parse-draft.md

```diff
@@ -0,0 +1,331 @@
+parse
+=====
+
+Use cases we identified:
+
+1. Parse an ISO 8601 string and get the exact information that is mandated by ISO 8601, which means that the whole string is relevant. (24:00 is midnight the following day, :60 seconds are coerced to :59).
+2. Parse something that might look like an ISO 8601 string but might have been generated incorrectly by some other program. In this case it's important to ignore irrelevant parts of the string.
+
+Here are some strings of interest, used in the examples below:
+```javascript
+// Valid string
+valid = '2001-09-08T18:46:40-07:00[America/Vancouver]';
+
+// String where the time affects the date
+time24 = '2020-04-07T24:00:00-07:00[America/Vancouver]';
+
+// Invalid time zone (although valid ISO 8601 string)
+invalidZone = '2001-09-08T18:46:40+01:00[America/Vancouver]';
+
+// Invalid date part, also invalid ISO 8601
+invalidDate = '2020-99-99T18:46:40-07:00[America/Vancouver]';
+
+// Invalid time part, also invalid ISO 8601
+invalidTime = '2001-09-08T99:99:99-07:00[America/Vancouver]';
+
+// Only date/time
+onlyDateTime = '2001-09-09T01:46:40';
+
+// Only date
+onlyDate = '2001-09-09';
+
+// In any of the proposals, these would only be accepted by YearMonth.from,
+// MonthDay.from, or Time.from, respectively
+onlyYearMonth = '2001-09';
+onlyMonthDay = '09-09';  // valid RFC 3339, I think technically not ISO 8601
```

Comment:
I don't think this comment is accurate; "MM-DD" is only ever a _part_ of a representation in either specification, not a full representation in its own right. The closest approximation I can find is in RFC 3339's [attempt to formalize ISO 8601 (1988)](https://tools.ietf.org/html/rfc3339#appendix-A), which includes `date = datespec-mday` and `datespec-mday = "--" dateopt-month date-mday`, allowing for values like "--09-09".

Reply:
You're right, I had misremembered this from when I was researching #404.